bin/install_npm_packages_clean

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -eu
+
+# Ensure package.json is present
+if [ ! -f package.json ]; then
+    echo "No package.json found!"
+    exit 1
+fi
+
+# Ensure package-lock.json is present
+# TODO: What to do with yarn-based projects? Make the hash on lockfile optional?
+if [ ! -f package-lock.json ]; then
+    echo "No package-lock.json found!"
+    exit 1
+fi
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+
+# TODO: This comes from Studio. Figure out if it's standard in Node or not.
+if [ -d patches ]; then
+  PATCHES_HASH=$(hash_directory patches/)
+else
+  PATCHES_HASH=nopatch
+fi
+
+CACHEKEY="$BUILDKITE_PIPELINE_SLUG-npm-$PLATFORM-$ARCHITECTURE-node-$NODE_VERSION-$PACKAGE_HASH-$PATCHES_HASH"
+
+LOCAL_NPM_CACHE=./vendor/npm
+mkdir -p $LOCAL_NPM_CACHE
+echo "--- :npm: Set npm to use $LOCAL_NPM_CACHE for cache"
+npm set cache $LOCAL_NPM_CACHE
+echo "npm cache set to $(npm get cache)"
+
+echo "--- :npm: Restore npm cache if present"
+restore_cache "$CACHEKEY"
+
+echo "--- :npm: Install Node dependencies"
+
+# To avoid constant ECONNRESET errors a limit is set for Linux,
+# as this is not happening with the Mac jobs.
+# This issue is being tracked here:
+# https://github.com/npm/cli/issues/4652
+if [ "$PLATFORM" = "Linux" ]; then
+  MAX_SOCKETS=1
+else
+  # Default value from npm.
+  # TODO: Link to source
+  MAX_SOCKETS=15
```

Comment:
The code this script was extracted from had the comment about 15 being the default. I had a _quick_ look and couldn't find a reference. Need to look more and document.